corehq/apps/userreports/expressions/specs.py

```diff
@@ -735,6 +735,86 @@ def _device_to_dict(
     return {k: v for k, v in device.to_json().items() if k in whitelist}
 
 
+class RelatedCaseExpressionSpec(JsonObject):
+    """
+    Similar to ``RelatedDocExpressionSpec``, this can be used to look up
```

Comment:
one thing that comes to mind is that related case is same as related doc for cases. I am not sure if its good to have two ways to do the same thing in long run. Or may be its fine to have a special expression for just cases?

nit: I'd recommend calling this `CaseExpressionSpec` since a relation with the root doc is really not necessary.

Reply:
> may be its fine to have a special expression for just cases?

Of course unlike all other doc types, which only have a doc_id, cases also have an external_id. That makes them special, and if we want to find related cases by their external_id, it makes sense to have a special expression for just cases.

> since a relation with the root doc is really not necessary

The `RelatedCaseExpressionSpec` works the same way as the `RelatedDocExpressionSpec`. For consistency, I reckon we either call them both `Related{Thing}ExpressionSpec` or both `{Thing}ExpressionSpec`.